wallet/client.go

```diff
@@ -118,11 +129,14 @@ func (c *grpcClient) sendTx(trx *tx.Tx) (tx.ID, error) {
 		return hash.UndefHash, err
 	}
 
+	ctx, cancel := context.WithTimeout(c.ctx, time.Second*5)
+	defer cancel()
+
 	data, err := trx.Bytes()
 	if err != nil {
 		return hash.UndefHash, err
 	}
-	res, err := c.transactionClient.BroadcastTransaction(c.ctx,
+	res, err := c.transactionClient.BroadcastTransaction(ctx,
```

Comment:
Important things is gRPC method `BroadcastTransaction` should pass context to lower layers to stop process.

When you call gRPC make a go routine in background to processing gRPC method then your context send signal cancel base on timeout, process don't canceled in background.

Reply:
Which layer I should pass it to?

Reply:
What should we do to cancel it on background? also, why it won't cancel in background when the timeout reaches?